File: kotlin-maven-playground/src/test/java/org/example/kotlin/core/java/ConstantsTest.java
package org.example.kotlin.core.java;

import com.example.kotlin.maven.playground.java.model.SomeAnnotation;
import com.example.kotlin.maven.playground.model.Constants;
import com.example.kotlin.maven.playground.model.ConstantsKt;
import org.junit.jupiter.api.Test;

public class ConstantsTest {

  @SomeAnnotation(text = ConstantsKt.CONST_VAL)
  public static class SomeAnnotatedClass {

  }

  @SomeAnnotation(text = Constants.CONST_VAL)
  public static class SomeAnnotatedClass2 {

  }

  @Test
  public void compileTimeConstants() {
    System.out.printf("annotation: %s%n",
        SomeAnnotatedClass.class.getAnnotation(SomeAnnotation.class).text());
    System.out.printf("annotation2: %s%n",
        SomeAnnotatedClass2.class.getAnnotation(SomeAnnotation.class).text());
  }

}

File: kotlin-maven-playground/src/test/kotlin/com/example/kotlin/maven/playground/kotlin/ConstantsTest.kt
package com.example.kotlin.maven.playground.kotlin

import com.example.kotlin.maven.playground.java.model.SomeAnnotation
import com.example.kotlin.maven.playground.model.CONST_VAL
import com.example.kotlin.maven.playground.model.Constants
import org.junit.jupiter.api.Test

class ConstantsTest {
  @SomeAnnotation(text = CONST_VAL)
  class SomeAnnotatedClass {
  }

  @SomeAnnotation(text = Constants.CONST_VAL)
  class SomeAnnotatedClass2 {
  }

  @Test
  fun compileTimeConstants() {
    println("annotation: ${SomeAnnotatedClass::class.java.getAnnotation(SomeAnnotation::class.java).text}")
    println("annotation2: ${SomeAnnotatedClass2::class.java.getAnnotation(SomeAnnotation::class.java).text}")
  }
}